### PR TITLE
Docs: Add UI components article.

### PR DESCRIPTION
**Pull request by gorzelinski**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Docs: Add UI components article. Closes https://github.com/cksource/ckeditor5-internal/issues/1898.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

**Comment by gorzelinski** [comment]
The article is almost ready. I'm still working on details, but I wanted to show what I have so far.
Screenshot 2023-03-02 at 16 20 06

File: docs/framework/architecture/ui-components.md [hunk 1]
```diff
@@ -0,0 +1,818 @@
+---
```
Comment (apadol-cksource) [inline docs/framework/architecture/ui-components.md:1]: Please add `modified_at` to add a "new" label.

**Review by apadol-cksource — CHANGES_REQUESTED** [review]
Great work and a great asset for documentation. I'm requesting few minor changes and I'm reporting 2 issues which may be related to the code and not the docs. To be verified.

File: docs/framework/architecture/ui-components.md [hunk 2]
```diff
@@ -0,0 +1,818 @@
...
+    class: 'ck-button-action'
+} );
+actionButton.render();
+
+document.getElementById( 'button-action' ).append( actionButton.element );
+```
+
+#### Rounded
+
+To get a rounded button, add the `ck-rounded-corners` class.
```
[the remark this answers is not in the document]
Comment (gorzelinski) [reply docs/framework/architecture/ui-components.md:105]: I was wondering why it's not displayed either. But, the class is applied - rounded corners only have 2px.
Screenshot 2023-03-06 at 10 14 04

File: docs/framework/architecture/ui-components.md [hunk 3]
```diff
@@ -0,0 +1,818 @@
...
+- `southEastArrowNorthWest`
+- `southEastArrowNorthMiddleWest`
+- `southEastArrowNorth`
+- `southEastArrowNorthMiddleEast`
+- `southEastArrowNorthEast`
+- `viewportStickyNorth`
+
+## Button
+
+{@snippet framework/ui/ui-button}
```
[the remark this answers is not in the document]
Comment (gorzelinski) [reply docs/framework/architecture/ui-components.md:76]: Yeah, it's a good idea.

**Review by apadol-cksource — APPROVED** [review]
LGTM

**Pull request merged**